### PR TITLE
new-feature/approval_reconcilation-api-endpoint

### PR DESCRIPTION
**Pull request by mikecarrier4**
# Pull Request Template

User Story

As an Admin-User I want to see the highest occurring and agreed upon location in the Admin Dashboard for an incident we've requested more information on. Determine how to handle consensus on suggestions for location information on incidents we request more information on via form.

Reasoning for the addition:  Admins want to review city , state, and time differences between different tweets that share the same incident_id.  Currently there is no feature with the admin dashboard that allows the admin to look at other tweets with the same incident_id.  This endpoints enables the front end team to be able to post incidents id and retrieve different data. 
1.) Get data on different incident_ids for admin review
pass action parameter 0
This allows the admin to look at different cities, states, and dates
2.)  Bring all tweet ids that associated with incident_id
pass action parameter 1
This allows the admin to see all tweet ids in the system that have the same incident id
3.) Bring the total number of incident ids in conversation table to the admin.
pass action parameter 2
This allows the admin to see count of rows with the same incident id

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

Locally and a new api end point will be added to the test file

**Test Configuration**:
* Firmware version:
* Hardware:
* Toolchain:
* SDK:

## Checklist:

- [X ] My code follows the style guidelines of this project
- [ X] I have performed a self-review of my own code
- [ X] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ X] My changes generate no new warnings
- [ X] I have added tests that prove my fix is effective or that my feature works
- [ X] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [ X] I have checked my code and corrected any misspellings

### CODE REVIEW

**Comment by mikecarrier4** [comment]
New API endpoint Test Here


curl -X 'POST' \
  'http://127.0.0.1:8000/approval_reconciliation/' \
  -H 'accept: application/json' \
  -H 'Content-Type: application/json' \
  -d '{
  "incident_id": 1224,
  "action": 0
}'

{
  "0": [
    {
      "count": 3,
      "root_tweet_city": "cHicago"
    },
    {
      "count": 1,
      "root_tweet_city": "foxlike"
    }
  ],
  "1": [
    {
      "count": 3,
      "root_tweet_state": "ILLInoIs"
    },
    {
      "count": 1,
      "root_tweet_state": "Wisconsin"
    }
  ],
  "2": [
    {
      "count": 4,
      "incident_date": "2020-05-30"
    }
  ]
}

**Comment by mikecarrier4** [comment]
curl -X 'POST' \
  'http://127.0.0.1:8000/approval_reconciliation/' \
  -H 'accept: application/json' \
  -H 'Content-Type: application/json' \
  -d '{
  "incident_id": 1224,
  "action": 1
}'

[
  {
    "tweet_id": "7832"
  },
  {
    "tweet_id": "9655585230177891645"
  },
  {
    "tweet_id": "6615007953146320971"
  },
  {
    "tweet_id": "2581507576373219659"
  }
]

**Comment by mikecarrier4** [comment]
curl -X 'POST' \
  'http://127.0.0.1:8000/approval_reconciliation/' \
  -H 'accept: application/json' \
  -H 'Content-Type: application/json' \
  -d '{
  "incident_id": 1224,
  "action": 2
}'
[
  {
    "anon_1": 4
  }
]

File: app/main.py [hunk 1]
```diff
@@ -72,6 +72,11 @@ class form_in(BaseModel):
 
 class check(BaseModel):
     tweet_id: str
+	
+	
+class check_action(BaseModel):
```
Comment (iknight7000) [inline app/main.py:77]: I can see where you added the new base model for this check_action.

**Review by iknight7000 — APPROVED** [review]
The admin should find this one useful when trying to decide on a location. One more step closer to automating more, less fun tasks.

File: app/main.py (hunk 1 above)
Comment (gaben3722) [reply app/main.py:77]: Good to check the base model. Approved

**Review by gaben3722 — APPROVED** [review]

**Pull request merged**